.github/workflows/build-on-main.yml

```diff
@@ -0,0 +1,100 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+##### README #####
+#
+# This CI action runs the release crates as downloaded from crates.io. It does not block PRs
+# and it is okay if it is flaky from time to time.
+
+name: Build Daily on Main
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    # Run this CI every day at 13 UTC / 5AM PST / 2PM CET
+    - cron:  '0 13 * * *'
+
+jobs:
+  test-cargo-cratesio:
+    name: "Test crates.io versions of ICU4X via the Cargo tutorials"
+    strategy:
+      matrix:
+        # TODO(#3109): Enable windows-latest.
+        os: [ ubuntu-latest, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+
+    # Cargo-make boilerplate
+    - name: Get cargo-make version
+      id: cargo-make-version
+      run: |
+        echo "hash=$(cargo search cargo-make | grep '^cargo-make =' | md5sum)" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Attempt to load cached cargo-make
+      uses: actions/cache@v3
+      id: cargo-make-cache
+      with:
+        path: |
+          ~/.cargo/bin/cargo-make
+          ~/.cargo/bin/cargo-make.exe
+        key: ${{ runner.os }}-make-${{ steps.cargo-make-version.outputs.hash }}
+    - name: Install cargo-make
+      if: steps.cargo-make-cache.outputs.cache-hit != 'true'
+      run: cargo +stable install cargo-make
+
+    # Toolchain boilerplate
+    - name: Potentially override rust version with nightly
+      run: cargo make set-nightly-version-for-ci
+    - name: Show the selected Rust toolchain
+      run: rustup show
+
+    # Actual job
+    - name: Run `cargo make test-cargo-tutorial`
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: make
+        args: test-cargo-tutorial-cratesio
+
+  test-cargo-local:
```

Comment:
It takes like 15 minutes. There's not much reason to run it because if it passes `cargo check`, it _probably_ runs all the way through. It's a lot of CI cycles for not much benefit. But I can add it back to build-test.yml if you prefer

Reply:
I'd prefer that.

Reply:
I changed the job back to the full build and changed it to debug mode. I'm also seeing whether it might be faster if we attach it to the end of one of the existing jobs.

Reply:
Runtimes are now:

- 8m 10s for a standalone job (1m 21s for setup and 6m 49s for the job)
- 5m 0s tacked onto the end of the testdata job
- 6m 14s tacked onto the end of the test job

I am therefore inclined to keep this as its own job.

Reply:
I wonder where those gains come from, it's a separate workspace so everything should be rebuilt.

Reply:
I believe the gains are from re-using the top level target directory to build a debug datagen binary, instead of cargo installing it.